examples/spring-boot-minimal/build.gradle

```diff
@@ -3,20 +3,27 @@ buildscript {
         gradlePluginPortal()
     }
     dependencies {
-        classpath "org.springframework.boot:spring-boot-gradle-plugin:${managedVersions['org.springframework.boot:spring-boot-gradle-plugin']}"
+        classpath "org.springframework.boot:spring-boot-gradle-plugin:${managedVersions['org.springframework.boot:spring-boot-dependencies']}"
+        classpath "io.spring.gradle:dependency-management-plugin:${managedVersions['io.spring.gradle:dependency-management-plugin']}"
     }
 }
 
 apply plugin: 'org.springframework.boot'
-
-springBoot {
-    mainClassName = 'example.springframework.boot.minimal.Main'
-}
+apply plugin: 'io.spring.dependency-management'
 
 dependencies {
     implementation project(':spring:boot-starter')
     implementation 'org.hibernate.validator:hibernate-validator'
 
+    implementation "com.spotify:completable-futures:${managedVersions['com.spotify:completable-futures']}"
+    implementation "it.unimi.dsi:fastutil:${managedVersions['it.unimi.dsi:fastutil']}"
+    implementation "com.google.guava:guava:${managedVersions['com.google.guava:guava']}"
+    implementation "com.google.code.findbugs:jsr305:${managedVersions['com.google.code.findbugs:jsr305']}"
+    implementation "org.jctools:jctools-core:${managedVersions['org.jctools:jctools-core']}"
+    implementation "org.reflections:reflections:${managedVersions['org.reflections:reflections']}"
+    implementation "org.bouncycastle:bcprov-jdk15on:${managedVersions['org.bouncycastle:bcprov-jdk15on']}"
+    implementation "javax.inject:javax.inject:${managedVersions['javax.inject:javax.inject']}"
```

Comment:
If I don't add these settings, can't compile due to the following error.
Seems If I add `dependency-management-plugin`, these settings are not applied.
https://github.com/line/armeria/blob/master/build.gradle#L73-L132

Reply:
from Boot 2.3.0 `productionRuntimeClasspath` is newly added and its `canBeConsumed` is `true`, so the version is not added because this condition is not meet. https://github.com/line/gradle-scripts/blob/master/lib/common-dependencies.gradle#L156
To solve this, I'm not sure which is the best. Perhaps we can set
`configurations.productionRuntimeClasspath.setCanBeConsumed(false)` if `bootJar` task exists.
@anuraaga Do you have any idea about this?

Reply:
By the way, this is how I managed to build this:
```
diff --git a/dependencies.yml b/dependencies.yml
index ef594b06c..1e216a11f 100644
--- a/dependencies.yml
+++ b/dependencies.yml
@@ -494,12 +494,16 @@ org.slf4j:
     - https://static.javadoc.io/org.slf4j/slf4j-api/1.7.30/
   slf4j-simple: { version: *SLF4J_VERSION }

+org.springframework:
+  spring-web: { version: '5.2.6.RELEASE' }
+
 org.springframework.boot:
   spring-boot-starter:
-    version: &SPRING_BOOT_VERSION '2.2.7.RELEASE'
+    version: &SPRING_BOOT_VERSION '2.3.0.RELEASE'
     javadocs:
     - https://docs.spring.io/spring/docs/current/javadoc-api/
   spring-boot-actuator-autoconfigure: { version: *SPRING_BOOT_VERSION }
+  spring-boot-autoconfigure: { version: *SPRING_BOOT_VERSION }
   spring-boot-starter-actuator: { version: *SPRING_BOOT_VERSION }
   spring-boot-starter-test: { version: *SPRING_BOOT_VERSION }
   spring-boot-starter-web: { version: *SPRING_BOOT_VERSION }
diff --git a/examples/spring-boot-minimal-kotlin/build.gradle.kts b/examples/spring-boot-minimal-kotlin/build.gradle.kts
index 59dd1b093..581f4014d 100644
--- a/examples/spring-boot-minimal-kotlin/build.gradle.kts
+++ b/examples/spring-boot-minimal-kotlin/build.gradle.kts
@@ -23,6 +23,9 @@ apply(plugin = "org.jetbrains.kotlin.plugin.spring")
 apply(plugin = "org.jlleitschuh.gradle.ktlint")
 apply(plugin = "org.springframework.boot")

+val productionRuntimeClasspath by configurations
+productionRuntimeClasspath.setCanBeConsumed(false)
+
 dependencies {
     implementation(project(":spring:boot-starter"))
     implementation("org.hibernate.validator:hibernate-validator")
diff --git a/examples/spring-boot-minimal/build.gradle b/examples/spring-boot-minimal/build.gradle
index bcf6a5621..4874c0019 100644
--- a/examples/spring-boot-minimal/build.gradle
+++ b/examples/spring-boot-minimal/build.gradle
@@ -13,6 +13,8 @@ springBoot {
     mainClassName = 'example.springframework.boot.minimal.Main'
 }

+configurations.productionRuntimeClasspath.setCanBeConsumed(false)
+
 dependencies {
     implementation project(':spring:boot-starter')
     implementation 'org.hibernate.validator:hibernate-validator'
diff --git a/examples/spring-boot-tomcat/build.gradle b/examples/spring-boot-tomcat/build.gradle
index 58e678673..848ff7de2 100644
--- a/examples/spring-boot-tomcat/build.gradle
+++ b/examples/spring-boot-tomcat/build.gradle
@@ -13,6 +13,8 @@ springBoot {
     mainClassName = 'example.springframework.boot.tomcat.Main'
 }

+configurations.productionRuntimeClasspath.setCanBeConsumed(false)
+
 dependencies {
     implementation project(':core')
     implementation project(':spring:boot-starter')
diff --git a/examples/spring-boot-webflux/build.gradle b/examples/spring-boot-webflux/build.gradle
index bf2e04e6c..151773407 100644
--- a/examples/spring-boot-webflux/build.gradle
+++ b/examples/spring-boot-webflux/build.gradle
@@ -13,6 +13,8 @@ springBoot {
     mainClassName = 'example.springframework.boot.webflux.Main'
 }

+configurations.productionRuntimeClasspath.setCanBeConsumed(false)
+
 dependencies {
     implementation project(':spring:boot-webflux-starter')

diff --git a/it/spring/boot-tomcat/build.gradle b/it/spring/boot-tomcat/build.gradle
index 0aa813e30..e4f60230b 100644
--- a/it/spring/boot-tomcat/build.gradle
+++ b/it/spring/boot-tomcat/build.gradle
@@ -4,5 +4,6 @@ dependencies {
     implementation('org.springframework.boot:spring-boot-starter-web') {
         exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
     }
+    implementation 'org.springframework:spring-web'
     testImplementation('org.springframework.boot:spring-boot-starter-test')
 }
diff --git a/it/spring/boot-tomcat8.5/build.gradle b/it/spring/boot-tomcat8.5/build.gradle
index 0d549447e..b89b07700 100644
--- a/it/spring/boot-tomcat8.5/build.gradle
+++ b/it/spring/boot-tomcat8.5/build.gradle
@@ -1,21 +1,28 @@
 final def SPRING_BOOT_VERSION = '2.1.14.RELEASE'
+final def SPRING_VERSION = '5.1.15.RELEASE'
 final def TOMCAT_VERSION = '8.5.55'

 dependencies {
     implementation project(':spring:boot-starter')
     implementation project(':tomcat8.5')
-    implementation("org.springframework.boot:spring-boot-starter") {
-        version {
-            // Will fail the build if the override doesn't work
-            strictly SPRING_BOOT_VERSION
+
+    [ 'spring-boot-starter', 'spring-boot-autoconfigure', 'spring-boot-starter-web' ].each {
+        implementation("org.springframework.boot:$it") {
+            version {
+                // Will fail the build if the override doesn't work
+                strictly SPRING_BOOT_VERSION
+            }
+            exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
         }
     }
-    implementation("org.springframework.boot:spring-boot-starter-web") {
-        version {
-            // Will fail the build if the override doesn't work
-            strictly SPRING_BOOT_VERSION
+
+    [ 'spring-beans', 'spring-web' ].each {
+        implementation("org.springframework:$it") {
+            version {
+                // Will fail the build if the override doesn't work
+                strictly SPRING_VERSION
+            }
         }
-        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
     }

     [ 'tomcat-embed-core', 'tomcat-embed-jasper', 'tomcat-embed-el' ].each {
diff --git a/spring/boot-actuator-autoconfigure/build.gradle b/spring/boot-actuator-autoconfigure/build.gradle
index 93c2aedc4..22fbd7fab 100644
--- a/spring/boot-actuator-autoconfigure/build.gradle
+++ b/spring/boot-actuator-autoconfigure/build.gradle
@@ -3,6 +3,8 @@ dependencies {
     api 'javax.inject:javax.inject'
     compileOnly 'javax.validation:validation-api'

+    implementation 'org.springframework.boot:spring-boot-autoconfigure'
+
     testImplementation 'org.springframework.boot:spring-boot-starter-actuator'
     testImplementation 'org.springframework.boot:spring-boot-starter-test'
     testImplementation 'io.projectreactor:reactor-test'
diff --git a/spring/boot-autoconfigure/build.gradle b/spring/boot-autoconfigure/build.gradle
index 3ae48457c..71d2b77d4 100644
--- a/spring/boot-autoconfigure/build.gradle
+++ b/spring/boot-autoconfigure/build.gradle
@@ -3,6 +3,8 @@ dependencies {
     compileOnly project(':thrift')
     implementation project(':logback')

+    implementation 'org.springframework.boot:spring-boot-autoconfigure'
+
     // TODO(anuraaga): Consider removing these since this module does not have related functionality.
     optionalApi 'io.micrometer:micrometer-registry-prometheus'
     optionalApi 'io.dropwizard.metrics:metrics-json'
```